### ISSUE TITLE
write job statistics to guest namespace on instance exit

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#5952]
Problem: we don't have much information about what happens in Flux instances launched by the system instance, unlike slurm which tracks job steps centrally.

Following up on a discussion in today's meeting about finding out what sort of "workflows" people are running:  Could we have a flux instance write a summary object of some sort to the guest KVS directory in its enclosing instance?

A tool could summarize those objects at the system instance level.

For extra credit they could be made hierarchical, e.g. the object would include a summary of the objects written by sub-instances in their KVS.

FWIW, [RFC 16](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_16.html) governs the content of the job KVS directory.

**Comment by grondo** [comment]
Good idea! One benefit of collecting that data periodically is that it would protect against instances that get killed due to timeout, exception etc.

A benefit of using annotations instead of a one time object in the kvs is that there's already a tool to present annotations at the instance level (`flux jobs`) and they are designed to be updated (as far as I recall) so you could get runtime stats as well. Once in the job-list database the data can be queried with `flux jobs` or whatever standard query tool we use instead of a custom tool.

The one-time summary in the kvs is appealing, however, because it is much simpler and it's easier to think about how to collect hierarchically as you suggested.

**Comment by grondo** [comment]
This is actually going to be quite important on systems where most or all jobs run at the system instance level are batch or alloc jobs. This is because most of the detail of the actual jobs run by the user is currently lost when the batch job exits, unless the user thought to use `--dump` on the command line.

For example it would be nice to be able to use `flux job taskmap` on a parallel job run within a batch job, or even be able to access job eventlogs. Perhaps we can persist some of this information in the KVS in such a way that some of these commands can work naturally.

**Comment by grondo** [comment]
@kkier brought this up again on today's coffee call.

The initial use case is just determining what _precisely_ was running on a given node at a time when some event occurs. Currently, sysadmins only know a given batch job was running at the time, but not what job within that job was running.

`--dump` as proposed in #6080 isn't a great solution here since having dumpfiles sprinkled through the fileystem isn't useful to the admins.

I wonder if a feature similar to `--dump`, but which writes to the guest namespace would be feasible? We could drop job output and just capture eventlogs and R. Maybe this information could be preserved as it occurs.
